modules/core/src/test/java/org/apache/ignite/internal/processors/cache/datastructures/IgniteCountDownLatchAbstractSelfTest.java

```diff
@@ -334,6 +339,58 @@ public void testLatchMultinode1() throws Exception {
     /**
      * @throws Exception If failed.
      */
+    public void testLatchBroadcast() throws Exception {
+        Ignite ignite = grid(0);
+        ClusterGroup srvsGrp = ignite.cluster().forServers();
+
+        int numOfSrvs = srvsGrp.nodes().size();
+
+        ignite.destroyCache("testCache");
+        IgniteCache<Object, Object> cache = ignite.createCache("testCache");
+
+        for(ClusterNode node: srvsGrp.nodes())
+            cache.put(String.valueOf(node.id()), 0);
+
+        for(int i=0; i<500; i++) {
+            IgniteCountDownLatch latch1 = createLatch1(ignite, numOfSrvs);
+            IgniteCountDownLatch latch2 = createLatch2(ignite, numOfSrvs);
+
+            ignite.compute(srvsGrp).broadcast(new IgniteRunnableJob(latch1, latch2, i));
+            assertTrue(latch2.await(10000));
+        }
+    }
+
+    /**
+     * @param client Ignite client.
+     * @param numOfSrvs Number of server nodes.
+     * @return Ignite latch.
+     */
+    private IgniteCountDownLatch createLatch1(Ignite client, int numOfSrvs) {
```

Comment:
Why do you need to create two different methods when the only difference between them is a latch name?

You can either create method like "createLatch1(Ignite client, int numOfSrvs, latchName)" or just inline creation of a latch into your test method. I prefer the second approach because presently this method body implementation is a one-liner.